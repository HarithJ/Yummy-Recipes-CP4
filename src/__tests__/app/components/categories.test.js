import React from 'react';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson, { shallowToJson } from 'enzyme-to-json';

import Categories from '../../../components/categories/categories.js'

describe("category module", () => {
    const categories = shallow(<Categories />)
    it("renders properly", () => {
      expect(shallowToJson(categories)).toMatchSnapshot();
    })

    it("has a function to get categories", () => {
      expect(categories.instance().getCategories());
    })

    it("has a function to set category to edit", () => {
      let categoryId = 2
      expect(categories.instance().setCategoryToEdit(categoryId));
    })

    it("has a function to call get categories", () => {
      expect(categories.instance().categoriesModified());
    })

    it("has a function to add a category", () => {
      expect(categories.instance().handleAddCategory(global.event));
    })

    it("has a function to set a search value", () => {
      categories.setProps({"history": {"push": jest.fn()}})
      expect(categories.instance().setSearchValue(global.event));
    })

    it("has a function to change a page", () => {
      let pageNumber = 2
      expect(categories.instance().changePage(pageNumber));
    })

    it("has a function to change the limit to number of categories", () => {
      let limitNumber = 3
      expect(categories.instance().changeLimit(limitNumber));
    })

    it("has a function to change the total number of pages", () => {
      expect(categories.instance().setTotalPageNumbers());
    })

    it("has a function to set alert msg", () => {
      expect(categories.instance().setAlertMsg());
    })

    it("has a function to hide alert", () => {
      expect(categories.instance().hideAlert());
    })
})
