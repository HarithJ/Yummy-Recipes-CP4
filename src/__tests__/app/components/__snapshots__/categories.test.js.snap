// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category module renders properly 1`] = `
<div>
  <Nav
    setSearchValue={[Function]}
  />
  <div
    className="container-fluid mt-3"
  >
    <br />
    <form
      action="/addcategory/"
      method="Post"
      name="addcategoryForm"
      onSubmit={[Function]}
    >
      <div
        className="row justify-content-center"
      >
        <div
          className="col-4"
        >
          <input
            className="form-control validate"
            name="category_name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="col-auto"
        >
          <button
            className="btn btn-success"
            type="submit"
          >
            Add Category
          </button>
        </div>
      </div>
    </form>
    <br />
  </div>
  <Pagination
    changeLimit={[Function]}
    changePage={[Function]}
    pagination={
      Object {
        "limit": 0,
        "page": 1,
        "totalPages": 1,
      }
    }
  />
</div>
`;
